internal/guest/commonutils/utilities.go

```diff
@@ -24,3 +33,74 @@ func DecodeJSONWithHresult(r io.Reader, v interface{}) error {
 	}
 	return nil
 }
+
+// StartTimeSyncService starts the `chronyd` deamon to keep the UVM time synchronized.  We
+// use a PTP device provided by the hypervisor as a source of correct time (instead of
+// using a network server). We need to create a configuration file that configures chronyd
+// to use the PTP device.  The system can have multiple PTP devices so we identify the
+// correct PTP device by verifying that the `clock_name` of that device is `hyperv`.
+func StartTimeSyncService() error {
+	ptpClassDir, err := os.Open("/sys/class/ptp")
+	if err != nil {
+		return errors.Wrap(err, "failed to open PTP class directory")
+	}
+
+	ptpDirList, err := ptpClassDir.Readdirnames(-1)
+	if err != nil {
+		return errors.Wrap(err, "failed to list PTP class directory")
+	}
+
+	var ptpDirPath string
+	found := false
+	for _, ptpDirPath = range ptpDirList {
+		clockNameFilePath := filepath.Join(ptpClassDir.Name(), ptpDirPath, "clock_name")
+		clockNameFile, err := os.Open(clockNameFilePath)
+		if err != nil {
+			return errors.Wrapf(err, "failed to open clock name file at %s", clockNameFilePath)
+		}
+		// Expected clock name is `hyperv` so read first 6 chars and verify the
+		// name
+		expectedReadLen := len("hyperv")
+		buf := make([]byte, expectedReadLen)
+		fileReader := bufio.NewReader(clockNameFile)
+		nread, err := fileReader.Read(buf)
+		if err != nil {
+			return errors.Wrapf(err, "read file %s failed", clockNameFilePath)
+		} else if nread != expectedReadLen {
+			return errors.Wrapf(err, "read file %s returned %d bytes, expected %d", clockNameFilePath, nread, expectedReadLen)
+		}
+		clockName := string(buf)
+		if strings.EqualFold(clockName, "hyperv") {
```

Comment:
I feel it makes more sense to have a string assigned "hyperv" and then check the length of the string on line 69, and then in the `strings.EqualFold` we can check against the same string for the second argument.